modules/axdriver/src/bus/mmio.rs

```diff
@@ -1,23 +1,69 @@
 #[allow(unused_imports)]
 use crate::{prelude::*, AllDevices};
+#[cfg(all(target_arch = "aarch64", feature = "irq"))]
+use arm_gic::gic_irq_translate;
 
 impl AllDevices {
     pub(crate) fn probe_bus_devices(&mut self) {
         // TODO: parse device tree
         #[cfg(feature = "virtio")]
-        for reg in axconfig::VIRTIO_MMIO_REGIONS {
-            for_each_drivers!(type Driver, {
-                if let Some(dev) = Driver::probe_mmio(reg.0, reg.1) {
-                    info!(
-                        "registered a new {:?} device at [PA:{:#x}, PA:{:#x}): {:?}",
-                        dev.device_type(),
-                        reg.0, reg.0 + reg.1,
-                        dev.device_name(),
-                    );
-                    self.add_device(dev);
-                    continue; // skip to the next device
+        {
+            #[cfg(any(
+                target_arch = "riscv32",
+                target_arch = "riscv64",
+                target_arch = "aarch64"
+            ))]
+            dtb::compatible_nodes("virtio,mmio", |n| {
+                let reg = n.prop("reg");
+                //todo: check size of data
+                let reg_base = reg.u64(0) as usize;
+                let reg_size = reg.u64(1) as usize;
+                #[cfg(feature = "irq")]
+                let irq_num;
+                #[cfg(feature = "irq")]
+                {
+                    #[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+                    {
+                        unimplemented!("No riscv's irq_num");
```

Comment:
rissv `irq_num` can be get by the "interrupts" property, too:

```
virtio_mmio@10001000 {
	interrupts = <0x01>;
	interrupt-parent = <0x03>;
	reg = <0x00 0x10001000 0x00 0x1000>;
	compatible = "virtio,mmio";
};
```

Reply:
IRQ for riscv has not been completed. It is better to do this before processing the irq number.